client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/use-pending-migration-status.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { useUpdateMigrationStatus } from 'calypso/data/site-migration/use-update-migration-status';
+import { HOW_TO_MIGRATE_OPTIONS } from 'calypso/landing/stepper/constants';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import type { NavigationControls } from '../../types';
+
+interface PendingMigrationStatusProps {
+	onSubmit?: Pick< NavigationControls, 'submit' >[ 'submit' ];
+}
+
+const usePendingMigrationStatus = ( { onSubmit }: PendingMigrationStatusProps ) => {
+	const site = useSite();
+	const siteId = site?.ID;
+
+	const canInstallPlugins = site?.plan?.features?.active.find(
+		( feature ) => feature === 'install-plugins'
+	)
+		? true
+		: false;
+
+	const { updateMigrationStatus } = useUpdateMigrationStatus();
+
+	// Register pending migration status when loading the step.
+	useEffect( () => {
+		if ( siteId ) {
+			updateMigrationStatus( siteId, 'migration-pending' );
+		}
+	}, [ siteId, updateMigrationStatus ] );
+
+	const onOptionClick = ( how: string ) => {
```

Comment:
I think we don't need to be opinionated here about if should be used as a click or any other way.
```suggestion
	const setPendingMigration = ( how: string ) => {
```

Reply:
Thank you for the review and suggestion, @gabrielcaires! Applied here: https://github.com/Automattic/wp-calypso/pull/95413/commits/4c8171449b01715a2e91a6cc8babf13f67410718